src/pocket/PocketSQSWithLambdaTarget.ts

```diff
@@ -15,26 +15,38 @@ import {
   IamRole,
   IamRolePolicyAttachment,
   LambdaEventSourceMapping,
+  DataAwsSqsQueue,
+  DataAwsSqsQueueConfig,
+  SqsQueue,
 } from '../../.gen/providers/aws';
 
 export interface PocketSQSWithLambdaTargetProps
   extends PocketVersionedLambdaProps {
-  sqsQueue?: {
-    messageRetentionSeconds?: number;
-    maxReceiveCount?: number;
-    maxMessageSize?: number;
-    delaySeconds?: number;
-    visibilityTimeoutSeconds?: number;
-  };
+  /**
+   * Set dataSqsQueue to use an existing SQS queue. If not provided, then a queue will be created.
+   */
+  configFromPreexistingSqsQueue?: DataAwsSqsQueueConfig;
```

Comment:
Could we call this `dataSqsQueue`? It conforms a little more to what terraform uses for existing resources even though the current name is very descriptive. I could be convinced otherwise :)

Reply:
I renamed this based on Chelsea's suggestion, but I don't have a strong opinion myself. In general, I like sticking to conventions, but the Terraform data prefix isn't a very good one, and we haven't adopted it in this repo yet.